### PR TITLE
Added initial integration testing framework

### PR DESCRIPTION
**Pull request by allenhsieh**
Added initial integration testing framework

This pull request makes the following changes:
* Moved projects into separate modules
* Set up nexus-repository-composer-it to use integration testing framework by @bhamail 
* Wrote basic tests to ensure framework is working

To test this locally run `mvn clean verify` you should see two tests run

### CODE REVIEW

File: nexus-repository-composer-it/src/test/it-resources/ftp-php.json [hunk 1]
```diff
@@ -0,0 +1 @@
..., easy-to-use library for accessing FTP servers.","keywords":[],"homepage":"","version":"v1.0.1","version_normalized":"1.0.1.0","license":[],"authors":[{"name":"David Grudl"},{"name":"Reinier Kip","email":"github@reinierkip.nl","role":"Developer"}],"dist":{"url":"http://localhost:8081/repository/composer-proxy/rjkip/ftp-php/v1.0.1/rjkip-ftp-php-v1.0.1.zip","type":"zip","reference":"8767a90886226bbfd9d12c61b0cf51ab171d814d","shasum":""},"type":"library","time":"2012-07-27T11:59:48+00:00","autoload":{"psr-0":{"FtpPhp\\":"src"}},"require":{"php":">=5.3"},"uid":15823},"v1.1.0":{"name":"rjkip/ftp-php","description":"A small, easy-to-use library for accessing FTP servers.","keywords":[],"homepage":"","version":"v1.1.0","version_normalized":"1.1.0.0","license":[],"authors":[{"name":"David Grudl"},{"name":"Reinier Kip","email":"github@reinierkip.nl","role":"Developer"}],"dist":{"url":"http://localhost:8081/repository/composer-proxy/rjkip/ftp-php/v1.1.0/rjkip-ftp-php-v1.1.0.zip","type":"zip","reference":"69869b3fe417172f3481e6245494843c31037366","shasum":""},"type":"library","time":"2014-01-13T19:24:51+00:00","autoload":{"psr-0":{"FtpPhp\\":"src"}},"require":{"php":">=5.3"},"uid":120076}}}}
```
Comment (DarthHater) [inline nexus-repository-composer-it/src/test/it-resources/ftp-php.json]: Trivial: newline

File: nexus-repository-composer-it/src/test/java/org/sonatype/nexus/repository/composer/internal/ComposerClient.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import org.sonatype.nexus.testsuite.testsupport.FormatClientSupport;
+
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.impl.client.CloseableHttpClient;
+
+public class ComposerClient
+    extends FormatClientSupport
+{
+  public ComposerClient(
+      final CloseableHttpClient httpClient,
```
Comment (DarthHater) [inline nexus-repository-composer-it/src/test/java/org/sonatype/nexus/repository/composer/internal/ComposerClient.java]: Formatting looks off here, pretty sure you can bring it up a line and align the rest of the params.

File: pom.xml [hunk 3]
```diff
@@ -1,99 +1,114 @@
...
 -->
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
     <parent>
         <groupId>org.sonatype.nexus.plugins</groupId>
         <artifactId>nexus-plugins</artifactId>
-        <version>3.13.0-01</version>
+        <version>3.19.0-SNAPSHOT</version>
```
Comment (DarthHater) [inline pom.xml]: @bhamail should this be `3.19.0-01` now that it's released?

File: pom.xml [hunk 4]
```diff
@@ -1,99 +1,114 @@
...
+        <module>nexus-repository-composer-it</module>
+    </modules>
+    <distributionManagement>
+        <snapshotRepository>
+            <id>ossrh</id>
+            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+        </snapshotRepository>
+    </distributionManagement>
+    <properties>
+        <nxrm-version>3.19.0-SNAPSHOT</nxrm-version>
```
Comment (DarthHater) [inline pom.xml]: Same query @bhamail , `3.19.0-01`?

File: pom.xml (hunk 3 above)
Comment (bhamail) [reply pom.xml]: Yes, in fact we can now use `3.19.1-01`.

File: pom.xml [hunk 5]
```diff
@@ -1,99 +1,114 @@
...
+            -->
+            <activation>
+                <property>
+                    <name>performRelease</name>
+                    <value>true</value>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <!--
```
Comment (bhamail) [inline pom.xml]: The `maven-enforcer-plugin` needs to be moved out of the \<profile> and into the "main" \<build> section, a la: https://github.com/sonatype-nexus-community/nexus-format-archetype/blob/master/src/test/resources/projects/it1/reference/pom.xml#L58

File: nexus-repository-composer/pom.xml [hunk 6]
```diff
@@ -0,0 +1,93 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.sonatype.nexus.plugins</groupId>
+    <artifactId>composer-parent</artifactId>
+    <version>0.0.2</version>
```
Comment (bhamail) [inline nexus-repository-composer/pom.xml]: Perhaps create an Issue in this project to track adding a "-SNAPSHOT" suffix to versioning (as was done in another project).

File: nexus-repository-composer-it/src/test/it-resources/ftp-php.json (hunk 1 above)
Comment (allenhsieh) [reply nexus-repository-composer-it/src/test/it-resources/ftp-php.json]: in: 5f4baf6

File: nexus-repository-composer-it/src/test/java/org/sonatype/nexus/repository/composer/internal/ComposerClient.java (hunk 2 above)
Comment (allenhsieh) [reply nexus-repository-composer-it/src/test/java/org/sonatype/nexus/repository/composer/internal/ComposerClient.java]: So strange, even though i have the sonatype codestyle, it was set to 4 spaces. Amazing eye.

In: 5f4baf6

File: pom.xml (hunk 3 above)
Comment (allenhsieh) [reply pom.xml]: in: 5f4baf6

File: pom.xml (hunk 4 above)
Comment (allenhsieh) [reply pom.xml]: in: 5f4baf6

File: pom.xml (hunk 5 above)
Comment (allenhsieh) [reply pom.xml]: in: 5f4baf6

File: nexus-repository-composer/pom.xml (hunk 6 above)
Comment (allenhsieh) [reply nexus-repository-composer/pom.xml]: Issue created: https://github.com/sonatype-nexus-community/nexus-repository-composer/issues/50

File: pom.xml [hunk 7]
```diff
@@ -1,99 +1,118 @@
...
+      <url>https://repository.sonatype.org/content/repositories/snapshots</url>
+    </repository>
+  </repositories>
+  <build>
+    <plugins>
+      <!--
+          Ensure use of Java 8, as required by NXRM
+          -->
+      <plugin>
+        <artifactId>maven-enforcer-plugin</artifactId>
```
Comment (bhamail) [inline pom.xml]: Thanks for this change. With this change in place, I now get the helpful error message below when I accidentally try to build with a "too new" java version (as opposed to an odd cryptic totally unrelated message):

```
[WARNING] Rule 0: org.apache.maven.plugins.enforcer.RequireJavaVersion failed with message:
Detected JDK Version: 11.0.4 is not in the allowed range [1.8,1.9).
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for org.sonatype.nexus.plugins:composer-parent-parent 0.0.2:
[INFO] 
[INFO] org.sonatype.nexus.plugins:composer-parent-parent .. FAILURE [  0.244 s]
```

**Pull request merged**